src/pages/tasks/common/hooks/useFetchInvoicesForTasksAdding.ts

```diff
@@ -0,0 +1,93 @@
+/**
+ * Invoice Ninja (https://invoiceninja.com).
+ *
+ * @link https://github.com/invoiceninja/invoiceninja source repository
+ *
+ * @copyright Copyright (c) 2022. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://www.elastic.co/licensing/elastic-license
+ */
+
+import { endpoint } from '$app/common/helpers';
+import { request } from '$app/common/helpers/request';
+import { route } from '$app/common/helpers/route';
+import { toast } from '$app/common/helpers/toast/toast';
+import { GenericSingleResourceResponse } from '$app/common/interfaces/generic-api-response';
+import { Invoice } from '$app/common/interfaces/invoice';
+import { Task } from '$app/common/interfaces/task';
+import { AxiosError } from 'axios';
+import { useQueryClient } from 'react-query';
+import { useSetAtom } from 'jotai';
+import { tasksToAddOnInvoiceAtom } from '../components/AddTasksOnInvoiceModal';
+import { Dispatch, SetStateAction } from 'react';
+
+interface Params {
+  setInvoices: Dispatch<SetStateAction<Invoice[]>>;
+  setIsAddTasksOnInvoiceVisible: Dispatch<SetStateAction<boolean>>;
+}
+
+export function useFetchInvoicesForTasksAdding(params: Params) {
+  const queryClient = useQueryClient();
+
+  const setTasksToAddOnInvoice = useSetAtom(tasksToAddOnInvoiceAtom);
+
+  const { setInvoices, setIsAddTasksOnInvoiceVisible } = params;
+
+  return (tasks: Task[]) => {
```

Comment:
@beganovich The idea of this hook is to use for both bulk and single task adding to invoice. So, that is the reason for accepting array instead of one task.